lib/ecto/repo/queryable.ex

```diff
@@ -307,10 +307,10 @@ defmodule Ecto.Repo.Queryable do
   defp process(row, {:struct, struct, args}, from, adapter) do
     {fields, row} = process_kv(args, row, from, adapter)
 
-    case Map.merge(struct.__struct__(), Map.new(fields)) do
+    case struct(struct, fields) do
```

Comment:
Perhaps we could go stricter and use `Kernel.struct!/2`?

Reply:
So the reason why it was written like above is because `struct` does a bunch of checking and in this case we are sure all of the fields belong to the struct, so we can skip the checking. It is done for performance reasons. :)

Reply:
I see! I will remove this then.